### ISSUE TITLE
Using tun interface (without ethernet layer)

### ISSUE THREAD

**Issue by JJTech0130** [ccie18643/PyTCP#30]
I've got a situation where it would be very useful to use pytcp as a TCP stack on top of a `tun`  interface, where the ethernet layer is not necessary. Was wondering how possible it would be to implement.

**Comment by ccie18643** [comment]
I don't think this would be big problem. Would need to add support for Tun interface and ensure the packet handler in such case works directly with IP layer. Feel free to hack a solution and send merge request if you have time. If not I'll implement this in version 3.0 (realistically will release it sometime in 2025).
